### ISSUE TITLE
please fix lint errors

### ISSUE THREAD

**Issue by coolgk** [mswjs/msw#249]
## Describe the bug
 example code should not have lint errors

![image](https://user-images.githubusercontent.com/4009549/85702621-804ce100-b6d6-11ea-990b-0344a876c014.png)

code 
```javascript
describe('Test', () => {
  const server = setupServer(
    rest.get('url', (_req, res, ctx) => {
      return res(ctx.text('abc'));
    }),
    rest.post('url', (req, res, ctx) => {
      return res(ctx.delay(1000), ctx.json({ ts: 2000 }));
    })
  );
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
});
```
tsconfig.json
```
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "strictNullChecks": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "baseUrl": ".",
    "noUnusedLocals": true
  },
  "exclude": ["node_modules"]
}
```
estlintrc
```
{
  "root": true,
  "extends": [
    "eslint:recommended",
    "plugin:security/recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
    "plugin:react/recommended",
    "plugin:jest/recommended",
    "plugin:jsx-a11y/recommended",
    "plugin:prettier/recommended"
  ],
  "env": {
    "es6": true,
    "node": true,
    "jest/globals": true
  },
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint", "security", "react", "jsx-a11y", "jest", "formatjs", "prettier"],
  "parserOptions": {
    "ecmaVersion": 8,
    "project": "./tsconfig.json",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "settings": {
    "react": {
      "version": "detect"
    }
  }
}
```

## Environment
packagejson

- `msw: 0.19.4`
- `nodejs: v12.13.0`
- `npm: 6.14.5`
"typescript": "^3.9.5"
"eslint": "^7.2.0",
    "eslint-config-prettier": "^6.11.0",
    "eslint-plugin-formatjs": "^2.3.13",
    "eslint-plugin-jest": "^23.13.2",
    "eslint-plugin-jsx-a11y": "^6.2.3",
    "eslint-plugin-prettier": "^3.1.3",
    "eslint-plugin-react": "^7.20.0",
    "eslint-plugin-security": "^1.4.0"

vscode
Version: 1.46.1 (system setup)
Commit: cd9ea6488829f560dc949a8b2fb789f3cdc05f5d
Date: 2020-06-17T21:13:20.174Z
Electron: 7.3.1
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363

## To Reproduce

open the code in vscode

## Expected behavior
example code should have no lint errors under the most strict rules

## Screenshots

![image](https://user-images.githubusercontent.com/4009549/85703432-44fee200-b6d7-11ea-8814-0194d2486310.png)

![image](https://user-images.githubusercontent.com/4009549/85703750-9018f500-b6d7-11ea-8f27-f2827551060f.png)

**Comment by kettanaito** [comment]
Hey, @coolgk. Not sure what is the issue here.

- If you are opening the `msw` source code (as a contributor, for example), you shouldn't see any lint errors, as it must use `eslint` config located in the repo.
- If you get lint errors from the `node_modules/msw` in _your project_, you should configure your lint setup to ignore `node_modules` (generally recommended).
- If you are copy-pasting the example code _into your project_, then it's up to you to adjust that code to suit your custom lint rules.

Does your use case fall into one of the aforementioned categories? Let me know if I misunderstood something. Thanks.

**Comment by coolgk** [comment]
thanks for the quick reply. I have reworded the first line. the example code in readme contains lint errors with strict typescript/estlint rules configured plus eslint-plugin-security. The lint errors are security-related. Could you provide examples of the typescript version of the examples?  I guess rest.get, rest.post might not return the correct types for settupWorkder/setupServer which caused lint errors. 

_If you are copy-pasting the example code into your project, then it's up to you to adjust that code to suit your custom lint rules._
we can't loosen rules just for msw...

**Comment by kettanaito** [comment]
I've checked if the `0.19.4` release ships with the correct type definitions, just in case they cannot be interpreted by your build. Here's the published state of `msw@0.19.4`:

- [`package.json`](https://unpkg.com/msw@0.19.4/package.json)
- [`lib/types/index.d.ts`](https://unpkg.com/msw@0.19.4/lib/types/index.d.ts)

> I can also see from the screenshots you've attached that the code does contain proper type definitions, so it's not the missing types issue.

You can see on your screenshot that `setupServer` returns an object with the public API properties (`listen`, `use`, etc.). Therefore, assigning `const server = setupServer()` is a totally safe assignment due to type inference. 

> we can't loosen rules just for msw...

Sorry, I phrased it wrong. I meant that you can edit the example code you've copied (it's just a regular code that lives in your code base now), and provide necessary adjustment so that your linter passes. Since lint rules are project-specific (there's no such thing as recommended security rules, despite [`@typescript-eslint/recommended`](https://github.com/mswjs/msw/blob/ebb85fb9f82c3e1bb0c117ebfc4b77e557d8cc61/.eslintrc.js#L8) name, which we use when building the library), it's virtually impossible to comply with an arbitrary set of rules that projects may have.

> I'm also confused why you get the [`no-unsafe-assignment`](https://github.com/typescript-eslint/typescript-eslint/blob/689dae37392d527c64ae83db2a4c3e6b7fecece7/packages/eslint-plugin/docs/rules/no-unsafe-assignment.md) rule violation, when the library doesn't export functions/methods typed as `any`.

My recommendation still goes with editing the code so it complies with your rules. If you spot that you cannot cover certain rules, because `msw` doesn't export certain types, let me know in this thread, we'll improve that!

**Comment by kettanaito** [comment]
Unless there are some missing types that we would have to export from the library, I'd consider this closed (see the motivation in the previous answer). Thanks for understanding. Please, feel free to use this thread to ask more questions, and generally share your progress in resolving this issue. I'd be glad to reopen, be there a need.
